### DISCUSSION TITLE
Automatic code formatting before commit

### DISCUSSION THREAD

**Discussion by lavafroth** [archlinux/archinstall#1662] [Ideas]
It would be pretty helpful if we setup a [pre-commit](https://pre-commit.com/) configuration with some code formatting tool like [black](https://github.com/psf/black) to maintain a uniform coding style among contributors.

**Comment by svartkanin** [comment]
There was a test for this done a while back using Yapf https://github.com/archlinux/archinstall/pull/997
But some formatting results weren't too popular :)

**Reply by lavafroth** [reply]
Agreed, Yapf might be more flexible than black. As far as I can see, the splitting of a line before a dictionary value can be controlled using the `ALLOW_SPLIT_BEFORE_DICT_VALUE` property. Also one dictionary key-value pair per line can be configured using `EACH_DICT_ENTRY_ON_SEPARATE_LINE`. I suppose, it Yapf might be a good candidate if we tweaked the knobs enough.

**Reply by Torxed** [reply]
I agree, I'm happy that @svartkanin introduced it to me. I've used it in some other projects as it's way more customizable and less intrusive if done right. Not perfect but it's the best alternative I think.

**Comment by Torxed** [comment]
Yea they're always insanely aggressive. And on paper sure they make sense but they're not very friendly for individual needs :)

A `sed s/    /\t/g` approach would have been epic hehe
